Documentation/learning/why.md

```diff
@@ -0,0 +1,21 @@
+# Why etcd
+
+etcd is designed to store a small amount of metadata in consistent, fault-tolerant way. Distributed systems use etcd as the root of scheduling and service discovery configuration storage. [Distributed lock][etcd-concurrency] is implemented on top of etcd, useful for master election in other distributed databases.
+
+# What does etcd stand for?
+
+TODO(fanmin)
+
+# Who uses etcd: container linux by CoreOS
+
+Application running on [container linux][container-linux] gets automatic, no-downtime Linux kernel updates. CoreOS uses [locksmith] to coordinate updates. locksmith stores semaphore values in etcd to ensure that only subset of cluster are rebooting at any given time.
```

Comment:
zero-downtime.

locksmith uses etcd to implement distributed semaphore, ensuring that ...